community/kernel/src/main/java/org/neo4j/storageengine/api/StorageEngine.java

```diff
@@ -113,6 +114,14 @@ void createCommands(
      */
     void prepareForRecoveryRequired();
 
+    /**
+     * As part of the simplified counts store. This will initialize the count store in the storage engine
+     * when doing recovery.
+     */
+    void initFromSnapshot( CountsSnapshot snapshot );
```

Comment:
Unfortunately I can't think of a better place right now.
These are other things that concern me:
 - counts store initialization grew through many layers, would be good to keep it more isolated
 - we expose not only counts store from `StorageEngine` but also the fact that it is initialized from the outside

Would be good to discuss this thing in person.